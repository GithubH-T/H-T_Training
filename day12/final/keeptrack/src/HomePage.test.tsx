import { render, screen } from "@testing-library/react"
import { createRenderer } from "react-dom/test-utils"
import HomePage from "./HomePage"
import renderer from 'react-test-renderer';

//Test suite-
//Test cases-
test("renders from heading", () => {
    render(<HomePage/>)
    expect(screen.getByText("Home")).toHaveTextContent("Home")
})
test("renders from heading", () => {
    render(<HomePage/>)
    expect(screen.getByRole("heading")).toHaveTextContent("Home")
})

//Add a snapshot test for the component. Organize the tests in a suite (describe function). Describe function- Missing second argument. It must be a callback function.

describe("<HPage/>",() => {
    test("should render w/o crashing", () => {
        render(<HomePage/>)
    })
    test("snapshot",()=> {
        const tree=renderer.create(<HomePage/>).toJSON();
        expect(tree).toMatchSnapshot()
    })
})