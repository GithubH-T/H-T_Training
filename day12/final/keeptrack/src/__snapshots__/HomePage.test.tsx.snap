// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HPage/> snapshot 1`] = `
<h2>
  Home
</h2>
`;
